test/jsx.jsx

```diff
@@ -125,4 +125,40 @@ test('name', () => {
     ]),
     'should support a fragment in an element (#2)'
   )
+
+  /**
+   * @typedef {import('../lib/core.js').HChild} HChild
+
+   * @param {{title: string; definition: string, children?: HChild}} options
+   * @returns {JSX.Element}
+   */
+  const Dl = ({title, definition, children}) => (
+    <dl>
+      <dt>{title}</dt>
+      <dd>{definition}</dd>
+      {children}
+    </dl>
+  )
```

Comment:
Bit of a weird test IMO, could a `DlEntry` with a `key` and `value` make more sense?

Would the types support something like `<DlEntry key={<b>something</b>} value={<p>Whatever</p>} />`?

Reply:
Yeah, I can give it a try. I also have now both the "jsxFactory" (classic) and "jsx transform" (automatic) forms working, so any help you can give to make sure that the tests are formed correctly and that the typings are correct would be greatly appreciated.